### PR TITLE
Addition of a ExtractMonitors

### LINKED ISSUE #17138: Create ExtractMonitorSpectra algorithm
This algorithm should behave similarly to `ExtractMonitorWorkspace`, but to be used where monitors and detectors are mixed within a workspace.

The algorithm should output the original workspace, with the same name, without monitors, and a new, separate workspace for the monitors.

The new monitor workspace should be set as the monitor workspace for the original workspace.

This allows the monitors to be separated, as for `LoadEventNexus`. However, for the ILL keeping the monitor workspace separate when initially loading causes problems for loading multiple files. This algorithm will also allow the monitor workspace to be separated at a chosen point (e.g. after loading or after background subtraction).

### PR DESCRIPTION
**Pull request by ianbush**
Adds a separate monitor spectra algorithm, which takes a workspace with detectors and monitors as input, and outputs a workspace with just detectors, and/or a workspace with just monitors.

**To test:**
- Load any file which creates detectors and monitors in a single workspace (e.g. LOQ49886.nxs files in usage data), and run the algorithm with different options set to test behaviour
- Load any file which creates detectors, and monitors in a linked workspace (e.g. CNCS_7860_event.nxs with LoadMonitors selected) and check an error is thrown, mentioning the name of the monitor workspace
- Can also test using CreateSampleWorkspace, with different numbers of monitors, detectors etc.
- Check documentation is clear

Fixes #17138.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by KarlPalmen** [comment]
I'm unassigning myself for the weekend to give someone else a chance to test this before the code freeze.

**Comment by KarlPalmen** [comment]
Test works as expected. I also tried it on the monitor workspace of the CNCS file and it worked for that producing both a monitor and detector workspace.

**Comment by KarlPalmen** [comment]
No mention of Release Notes found.

File: Framework/PythonInterface/plugins/algorithms/SeparateMonitorSpectra.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+from __future__ import (absolute_import, division, print_function)
+from mantid.simpleapi import *
+from mantid.kernel import *
+from mantid.api import (MatrixWorkspaceProperty, DataProcessorAlgorithm, PropertyMode)
+
+class SeparateMonitorSpectra(DataProcessorAlgorithm):
```
Comment (martyngigg) [inline Framework/PythonInterface/plugins/algorithms/SeparateMonitorSpectra.py]: This sounds like a useful addition but I think a name such as `ExtractMonitors` or `ExtractMonitorSpectra` (my preference would be for the shorter first one) would be better. We tend to use the work `extract` over `separate`.

**Comment by ianbush** [comment]
I have gone with Martin's suggestion of renaming this to ExtractMonitors. This does seem to fit in more with other algorithms, and ~~seperate~~ separate is harder to spell!

I have also added the missing release notes.

**Comment by martyngigg** [comment]
@KarlPalmen Could you retest this now?

**Comment by KarlPalmen** [comment]
I'm building for #16855 at the moment.

File: docs/source/release/v3.8.0/framework.rst [hunk 2]
```diff
@@ -87,6 +89,12 @@ Improved
 - :ref:`ConvertUnits <algm-ConvertUnits>` now has the option to take a workspace with Points as input.
   A property has been added that will make the algorithm convert the workspace to Bins automatically. The output space will be converted back to Points.
 
+- :ref:`ConvertToHistogram <algm-ConvertToHistogram>`: Performance improvement using new HistogramData module,
```
Comment (AndreiSavici) [inline docs/source/release/v3.8.0/framework.rst]: Why are these release notes in this PR?

**Comment by KarlPalmen** [comment]
Test works as expected. As stated it no longer works with the CNCS example and the name of the monitor workspace is shown in the error message. :shipit:

File: docs/source/release/v3.8.0/framework.rst (hunk 2 above)
Comment (KarlPalmen) [inline docs/source/release/v3.8.0/framework.rst]: My guess is erroneous issue number for a commit for another issue.

Comment (martyngigg) [inline docs/source/release/v3.8.0/framework.rst]: @ianbush Can you remove this line in the release notes?

Comment (ianbush) [inline docs/source/release/v3.8.0/framework.rst]: I am not sure how they got there, I must have made a mistake when merging. Fixed now.

**Pull request merged**